lintcheck/src/json.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::HashMap;
+use std::fmt::Write;
+use std::fs;
+use std::hash::Hash;
+use std::path::Path;
+
+use crate::ClippyWarning;
+
+/// Creates the log file output for [`crate::config::OutputFormat::Json`]
+pub(crate) fn output(clippy_warnings: &[ClippyWarning]) -> String {
+    serde_json::to_string(&clippy_warnings).unwrap()
+}
```

Comment:
It's possible but yeah more work, currently the JSON mode will print the ICEs + panic which would fail CI (though not block bors, I think) before it gets to the `diff` step. I think that would work well enough since we wouldn't want to accept any change that introduces ICEs if we spot them

Reply:
> currently the JSON mode will print the ICEs + panic which would fail CI

Okay, that will be good enough. I think having it all uniformly outputted would make it easier to review and later potential bot integration. But that shouldn't block this PR

> (though not block bors, I think) 

Correct, Bors runs a different `.yaml` so this should be fine